elasticsearch/client.go

```diff
@@ -18,81 +18,88 @@
 package elasticsearch
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
+	"errors"
 	"io"
+	"io/ioutil"
 	"net/http"
+	"net/url"
+	"strings"
 
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/version"
 
 	v7 "github.com/elastic/go-elasticsearch/v7"
-	v7esapi "github.com/elastic/go-elasticsearch/v7/esapi"
 
 	v8 "github.com/elastic/go-elasticsearch/v8"
-	v8esapi "github.com/elastic/go-elasticsearch/v8/esapi"
 )
 
 // Client is an interface designed to abstract away version differences between elasticsearch clients
 type Client interface {
+	// TODO: deprecate
 	// Search performs a query against the given index with the given body
 	Search(index string, body io.Reader) (int, io.ReadCloser, error)
-	SecurityHasPrivilegesRequest(body io.Reader, header http.Header) (int, io.ReadCloser, error)
+	// Makes a request with application/json Content-Type and Accept headers by default
+	// pass/overwrite headers with "key: value" format
+	JSONRequest(method, path string, body interface{}, headers ...string) JSONResponse
 }
 
 type clientV8 struct {
-	client *v8.Client
+	c *v8.Client
 }
 
 // Search satisfies the Client interface for version 8
-func (c clientV8) Search(index string, body io.Reader) (int, io.ReadCloser, error) {
-	return v8Response(c.client.Search(
-		c.client.Search.WithContext(context.Background()),
-		c.client.Search.WithIndex(index),
-		c.client.Search.WithBody(body),
-		c.client.Search.WithTrackTotalHits(true),
-		c.client.Search.WithPretty(),
-	))
-}
-
-func (c clientV8) SecurityHasPrivilegesRequest(body io.Reader, header http.Header) (int, io.ReadCloser, error) {
-	hasPrivileges := v8esapi.SecurityHasPrivilegesRequest{Body: body, Header: header}
-	return v8Response(hasPrivileges.Do(context.Background(), c.client))
-}
-
-func v8Response(response *v8esapi.Response, err error) (int, io.ReadCloser, error) {
+func (v8 clientV8) Search(index string, body io.Reader) (int, io.ReadCloser, error) {
```

Comment:
> you pushed for not using abbreviations except for temporary variables

Not true. I did push for best judgement thou! - not to take everything I say as a hammer/strict rule for every case! (you know I'm not a person of hard rules)
In the very controversial topic of abbreviations, I was talking about not having to think & dig what a non standard abbreviation means, particularly when the context doesn't particularly help... Which again is a subjective topic and I realise it might not be perfectly clear here, so I flipped it nevertheless to see if it helps.

Reply:
I remember you claimed we should use `index` over `idx` etc. IMO this is pushing for not using abbreviations. 
But even happier when I misunderstood, as in most cases I am in favour of abbreviations as long as they are not cryptic. Seems we are on the same page then.